src/quo/components/share/share_qr_code/view.cljs

```diff
@@ -1,46 +1,243 @@
-       :underlay-color   colors/neutral-80-opa-1-blur
-       :background-color :transparent
-       :on-press         url-on-press
-       :on-long-press    url-on-long-press
-       :style            style/profile-address-content-container}
-      [text/text
-       {:style           style/profile-address-content
-        :size            :paragraph-1
-        :weight          :medium
-        :ellipsize-mode  :middle
-        :number-of-lines 1}
-       qr-url]]]
-    [rn/view {:style style/share-button-container}
-     [button/button
-      {:icon-only?          true
-       :type                :grey
-       :background          :blur
-       :size                32
-       :accessibility-label :share-profile
-       :on-press            share-on-press}
-      :i/share]]]])
+  "Receives the following properties:
+   - type:                :profile | :wallet-legacy | :wallet-multichain
+   - qr-image-uri:        Image source value.
+   - qr-data:             Text to show below the QR code.
+   - on-text-press:       Callback for the `qr-data` text.
+   - on-text-long-press:  Callback for the `qr-data` text.
+   - on-share-press:      Callback for the share button.
+   - customization-color: Custom color for the QR code component.
+   - unblur-on-android?:  [Android only] disables blur for this component.
+
+   Depending on the `type`, different properties are accepted:
+   `:profile`
+     - full-name:       User full name.
+     - profile-picture: map ({:source image-source}) or any image source.
+   `:wallet-legacy`
+     - emoji:               Emoji in a string to show in the QR code.
+     - on-info-press:       Callback for the info icon.
+     - on-legacy-press:     Callback for the legacy tab.
+     - on-multichain-press: Callback for the multichain tab.
+   `:wallet-multichain`
+     - networks:            A vector of network names as keywords (`[:ethereum, :my-net, ...]`).
+     - on-settings-press:   Callback for the settings button.
+     - emoji:               Emoji in a string to show in the QR code.
+     - on-info-press:       Callback for the info icon.
+     - on-legacy-press:     Callback for the legacy tab.
+     - on-multichain-press: Callback for the multichain tab."
+  [{:keys [unblur-on-android?] :as props}]
+  (reagent/with-let [component-width     (reagent/atom nil)
+                     container-component (if (and platform/android? unblur-on-android?)
+                                           [rn/view {:background-color style/overlay-color}]
+                                           [blur/view
+                                            {:blur-radius      20
+                                             :blur-amount      20
+                                             :blur-type        :transparent
+                                             :overlay-color    style/overlay-color
+                                             :background-color style/overlay-color}])]
+    [quo.theme/provider {:theme :dark}
```

Comment:
I don't think we need the theme provider hardcoded here? It's just a case the we don't care about Thai component in light theme as we never use that

Reply:
Sorry, I don't get the idea.

I added this because all the styles and subcomponents in this component are expected to be rendered in dark mode regardless the current app theme, for example: texts, tabs and buttons.

I'm unsing this component in the preview screen (which can be dark or light), and we could use this component in any screen, so I think setting the `theme/provider` as dark is a task of this component.

Reply:
Hmm, we have not done this for other components. The point here is that we will never use this component in light theme anyway and even if someone is looking at the preview screens of light theme well they are looking at an incorrect configuration. Ideally tooling like Malli will handle this in the preview screens (maybe show an icon when using a misconfiguration or something ).

I'm not sure we need to have this type of guard railing on the components.

Reply:
I used this approach here for the Scan QR screen  https://github.com/status-im/status-mobile/blob/5829eaf77b5aee484015ddc1868c78d37a2ef812/src/status_im2/common/scan_qr_code/view.cljs#L229
As you can see here https://github.com/status-im/status-mobile/blob/6bb8f1204159433c20335b65e3cece959913c250/src/status_im2/navigation/screens.cljs#L275 the screen is forced to be dark, but the button should be light as per Figma designs https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=689-193111&mode=design&t=XRiavTx7gYh1t2PG-4

I can see other parts of the codebase where we are using the same approach, like here https://github.com/status-im/status-mobile/blob/6f9bcd1bb1754f7735d33e966717c82e32415c2f/src/status_im2/contexts/shell/jump_to/components/bottom_tabs/view.cljs#L63

Reply:
Just an fyi - this example

https://github.com/status-im/status-mobile/blob/6f9bcd1bb1754f7735d33e966717c82e32415c2f/src/status_im2/contexts/shell/jump_to/components/bottom_tabs/view.cljs#L63

Is a bit different because it's a chunk of the screen which is in dark theme regardless of the theme of the page. I.e that's the bottom tabs in the application which is always in dark but the screen above it can be light/dark.

In the case of this component here- it will only be used on a page/wrapping with dark theme.

In any case I'm not strongly against the approach